specifyweb/stored_queries/execution.py

```diff
@@ -441,7 +441,7 @@ def execute(session, collection, user, tableid, distinct, count_only, field_spec
     "Build and execute a query, returning the results as a data structure for json serialization"
 
     set_group_concat_max_len(session)
-    query, order_by_exprs = build_query(session, collection, user, tableid, field_specs, recordsetid=recordsetid, formatauditobjs=formatauditobjs)
+    query, order_by_exprs = build_query(session, collection, user, tableid, field_specs, recordsetid=recordsetid, formatauditobjs=formatauditobjs, distinct=distinct)
 
     if distinct:
```

Comment:
Should the responsibility for making the query distinct be moved into `build_query` since it knows about that option now?